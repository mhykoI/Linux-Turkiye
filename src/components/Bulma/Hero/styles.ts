import { Styles } from 'react-jss';

export default {
  hero: {
    background: 'transparent',
  },
} as Styles;
