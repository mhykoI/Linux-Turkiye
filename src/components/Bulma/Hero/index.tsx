import React from 'react';
import clsx from 'clsx';
import { createUseStyles } from 'react-jss';
import Container from '../Container';
import { Bulma } from '../../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface HeroProps {
  size?: Bulma.HeroSize;
  particles?: boolean;
  heroHead?: React.ReactNode;
  heroBody?: React.ReactNode;
  heroFoot?: React.ReactNode;
}

const Hero: React.FC<HeroProps> = ({ size, heroHead, heroBody, heroFoot }) => {
  const classes = useStyles();

  return (
    <section className={clsx('hero', classes.hero, { [`is-${size}`]: size })}>
      <div className="hero-head">{heroHead}</div>

      <div className="hero-body">
        <Container className="has-text-centered">{heroBody}</Container>
      </div>

      <div className="hero-foot">{heroFoot}</div>
    </section>
  );
};

export default Hero;
