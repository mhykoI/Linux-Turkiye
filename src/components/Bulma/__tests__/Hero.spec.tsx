import React from 'react';
import renderer from 'react-test-renderer';
import Hero from '../Hero';

describe('<Hero />', () => {
  test('should render correctly', () => {
    const heroHead = <h1>Hero Head</h1>;
    const heroBody = <div>Hero Body</div>;
    const heroFoot = <span>Hero Footer</span>;

    const tree = renderer
      .create(
        <Hero
          heroHead={heroHead}
          heroBody={heroBody}
          heroFoot={heroFoot}
          particles
          size="medium"
        >
          This is a Box!
        </Hero>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
