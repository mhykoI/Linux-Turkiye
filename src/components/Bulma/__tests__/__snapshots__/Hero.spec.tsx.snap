// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render correctly 1`] = `
<section
  className="hero hero-0-2-1 is-medium"
>
  <div
    className="hero-head"
  >
    <h1>
      Hero Head
    </h1>
  </div>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div>
        Hero Body
      </div>
    </div>
  </div>
  <div
    className="hero-foot"
  >
    <span>
      Hero Footer
    </span>
  </div>
</section>
`;
